src/main/kotlin/uk/gov/justice/digital/hmpps/sentenceplan/services/ARNSApiService.kt

```diff
@@ -18,19 +19,24 @@ import uk.gov.justice.digital.hmpps.sentenceplan.stub.StubData
 
 @Service
 class ARNSApiService(
-  private val arnsRestClient: ARNSRestClient,
-  private val deliusRestClient: DeliusRestClient,
+  @Qualifier("arnsRestClient") private val arnsRestClient: WebClient,
+  @Qualifier("deliusRestClient")private val deliusRestClient: WebClient,
   @Value("\${use-stub}") private val useStub: Boolean,
 ) {
 
   fun getPopInfo(crn: String): PopInfoResponse {
     val caseDetail: CaseDetail =
       if (useStub) {
         log.info("Calling Stub")
-        StubData.getCaseDetail(crn) ?: throw ResponseStatusException(HttpStatus.NOT_FOUND)
+        StubData.getCaseDetail(crn)
       } else {
         log.info("Calling DeliusRestClient")
-        deliusRestClient.getCaseDetail(crn) ?: throw ResponseStatusException(HttpStatus.NOT_FOUND)
+        deliusRestClient.get()
+          .uri("/case-details/$crn")
+          .retrieve()
+          .bodyToMono(CaseDetail::class.java)
+          .onErrorResume(WebClientResponseException.NotFound::class.java) { throw ResponseStatusException(HttpStatus.NOT_FOUND) }
+          .block()
```

Comment:
is the behaviour asynchronous without the block()?

Reply:
It can be if we change the return types of your service methods and refactor the controllers to handle reactive types. The way it's set up though with a call into the API which then calls an external api, not sure it would help much.

Reply:
yes, wasn't suggesting we do, just clarifying for own purposes, thanks!